bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -379,27 +373,27 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 		return new Point(width, height);
 	}
 
+	private boolean shouldDrawCloseIcon(CTabItem item) {
+		CTabFolder folder = item.getParent();
+		boolean showClose = folder.showClose || item.showClose;
+		boolean isSelectedOrShowCloseForUnselected = (item.state & SWT.SELECTED) != 0 || folder.showUnselectedClose;
+		return showClose && isSelectedOrShowCloseForUnselected;
+	}
+
 	/**
-	 * Returns padding for the text of a tab when image is not available or is hidden.
-	 *
-	 * @param item CTabItem
-	 * @param state current state
-	 *
+	 * Returns padding for the text of a tab item when showing images is disabled for the tab folder.
 	 */
-	private int getTextPadding(CTabItem item, int state) {
+	private int getLargeTextPadding(CTabItem item) {
 		CTabFolder parent = item.getParent();
 		String text = item.getText();
 
 		if (text != null && parent.getMinimumCharacters() != 0) {
-			if (applyLargeTextPadding(parent)) {
-				return TABS_WITHOUT_ICONS_PADDING;
-			}
+			return TABS_WITHOUT_ICONS_PADDING;
 		}
-
 		return 0;
 	}
 
-	private boolean applyLargeTextPadding(CTabFolder tabFolder) {
+	private boolean shouldApplyLargeTextPadding(CTabFolder tabFolder) {
```

Comment:
That's true, but that is the behavior even without the changes in this PR. And in particular, that is the behavior that has been existing for years and must not be changed.
The padding-based separation of tabs, as introduced with #785, is only active if images are _completely_ disabled for tabs. This is currently identified by both showing images of selected and unselected tabs is set to false, as the former was always true before #785, so that this way of configurationidentification did not produce regressions. Still, we should think of more explicit approach to apply text-based padding to a CTabFolder (see https://github.com/eclipse-platform/eclipse.platform.swt/pull/785#issuecomment-1858837492).

Reply:
In the context of avoiding regression it makes sense to not apply this suggestion. Will evaluate this on #945 👍